### PR TITLE
[PIPELINES 3] Add second pipeline deploying from tagged image

### PR DESCRIPTION
**Pull request by petrkotas**
### Which issue this PR addresses:

Fixes [9741850](https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/9741850/)

### What this PR does / why we need it:

Introduces a copy of deploy pipelines, but utilizes tag to deploy.
Tagged image is pulled and `aro` binary is extracted and placed into the appropriate directory.
The rest of the pipeline is the same.

Contains: #1674

Signed-off-by: Petr Kotas <pkotas@redhat.com>

### Test plan for issue:
Manual

### Is there any documentation that needs to be updated for this PR?

Doc will be added as a whole.

### CODE REVIEW

File: .pipelines/deploy-azure-env-tag.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+trigger: none
+pr: none
+
+parameters:
+- name: location
+- name: vsoConfigBuildID
+- name: aroImageTag
+- name: rpImageAcr
+  type: string
+  default: arosvc
```
Comment (petrkotas) [inline .pipelines/deploy-azure-env-tag.yml:11]:
```suggestion
  - name: aroImageTag
  - name: rpImageAcr
    type: string
    default: arosvc
```
New parameters added to controll the deployment

Comment (petrkotas) [inline .pipelines/deploy-azure-env-tag.yml:11]:
```
  - name: aroImageTag
  - name: rpImageAcr
    type: string
    default: arosvc
```
New parameters added to control the deployment

File: .pipelines/deploy-azure-env-tag.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    azureDevOpsJSONSPN: $(aro-v4-ci-devops-spn)
+    vsoProjectID: $(vso-project-id)
+    vsoConfigPipelineID: $(vso-config-pipeline-id)
+    vsoConfigBuildID: ${{ parameters.vsoConfigBuildID }}
+    azureDevOpsE2EJSONSPN: $(aro-v4-e2e-devops-spn)
+    e2eSubscription: $(e2e-subscription)
+    billingE2EPipelineName: $(billing-e2e-pipeline-name)
+    billingE2EBranchName: $(billing-e2e-branch-name)
+    tag: ${{ parameters.aroImageTag }}
+    rpImageAcr: ${{ parameters.rpImageAcr }}
```
Comment (petrkotas) [inline .pipelines/deploy-azure-env-tag.yml:31]: params used here

File: .pipelines/int-release-tag.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+trigger: none
+
+parameters:
+- name: vsoConfigBuildID
+  type: string
+  default: latest
+- name: rpImageAcr
+  type: string
+  default: arosvc
+- name: aroImageTag
```
Comment (petrkotas) [inline .pipelines/int-release-tag.yml:12]: New parameters added to control the deployment

File: .pipelines/int-release-tag.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+stages:
+- stage: Deploy_INT
+  dependsOn: []
+  displayName: 🚀 Deploy INT
+  jobs:
+  - template: ./templates/template-job-deploy-azure-env-tag.yml
```
Comment (petrkotas) [inline .pipelines/int-release-tag.yml:19]: switched to new template

File: .pipelines/int-release-tag.yml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+      azureDevOpsJSONSPN: $(aro-v4-ci-devops-spn)
+      vsoProjectID: $(vso-project-id)
+      vsoConfigPipelineID: $(vso-config-pipeline-id)
+      vsoConfigBuildID: ${{ parameters.vsoConfigBuildID }}
+      azureDevOpsE2EJSONSPN: $(aro-v4-e2e-devops-spn)
+      e2eSubscription: $(e2e-subscription)
+      billingE2EPipelineName: $(billing-e2e-pipeline-name)
+      billingE2EBranchName: $(billing-e2e-branch-name)
+      tag: ${{ parameters.aroImageTag }}
+      rpImageAcr: ${{ parameters.rpImageAcr }}
```
Comment (petrkotas) [inline .pipelines/int-release-tag.yml:36]: use new paramaters

File: .pipelines/prod-release-tag.yml [hunk 6]
```diff
@@ -0,0 +1,154 @@
+# No PR triggers to run it manually
+pr: none
+trigger: none
+
+parameters:
+- name: vsoConfigBuildID
+- name: aroImageTag
+- name: rpImageAcr
+  type: string
+  default: arosvc
```
Comment (petrkotas) [inline .pipelines/prod-release-tag.yml:10]: New parameters added to control the deployment

File: .pipelines/prod-release-tag.yml [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+
+stages:
+- stage: Deploy_CanarySector
+  dependsOn: []
+  displayName: 🚀 Deploy Canary Sector
+  jobs:
+  - template: ./templates/template-job-deploy-azure-env-tag.yml
```
Comment (petrkotas) [inline .pipelines/prod-release-tag.yml:17]: switched to new template

File: .pipelines/templates/template-job-deploy-azure-env-tag.yml [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+  configFileName: ''
+  e2eSubscription: ''
+  environment: ''
+  locations: []
+  rpMode: ''
+  vsoProjectID: ''
+  vsoConfigPipelineID: ''
+  vsoConfigBuildID: ''
+  tag: ''
+  rpImageAcr: ''
```
Comment (petrkotas) [inline .pipelines/templates/template-job-deploy-azure-env-tag.yml:17]: New template paramaters

File: .pipelines/templates/template-job-deploy-azure-env-tag.yml [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+              export TAG=${{ parameters.vsoProjectID }}
+              export RP_IMAGE_ACR=${{ parameters.rpImageAcr }}
+
+              az acr login --name "${RP_IMAGE_ACR}"
+
+              docker pull "${RP_IMAGE_ACR}.azurecr.io/aro:${TAG}"
+              CONTAINER=$(docker create "${RP_IMAGE_ACR}.azurecr.io/aro:${TAG}")
+              docker cp $CONTAINER:/usr/local/bin/aro $(System.ArtifactsDirectory)/deployer/aro
+            displayName: Download aro deployer from image
+          - template: ./templates/template-az-cli-logout.yml
```
Comment (petrkotas) [inline .pipelines/templates/template-job-deploy-azure-env-tag.yml:62]: Replaces pull from pipeline artifact store. Uses parameters to configure the pull

**Review by bryanro92 — APPROVED** [review]

File: .pipelines/templates/template-job-deploy-azure-env-tag.yml [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+    environment: ${{ parameters.environment }}
+    strategy:
+      runOnce:
+        deploy:
+          steps:
+          - ${{ if eq(parameters.imageTag, 'latest') }}:
+            - checkout: git://AzureRedHatOpenShift/RP-Config@refs/heads/master
+              path: $(Agent.BuildDirectory)/s/config
+          - ${{ if ne(parameters.imageTag, 'latest') }}:
+            - checkout: git://AzureRedHatOpenShift/RP-Config@refs/tags/${{parameters.imageTag}}
```
Comment (petrkotas) [inline .pipelines/templates/template-job-deploy-azure-env-tag.yml:34]: @bennerv I fixed the parameter names, now only `imageTag` is used everywhere.

File: .pipelines/templates/template-job-deploy-azure-env-tag.yml [hunk 11]
```diff
@@ -0,0 +1,119 @@
+# Azure DevOps Job deploying rp
+parameters:
+  aroVersionStorageAccount: ''
+  azureDevOpsE2EJSONSPN: ''
+  azureDevOpsJSONSPN: ''
+  billingE2EPipelineName: ''
+  billingE2EBranchName: ''
+  configVersion: ''
```
Comment (bennerv) [inline .pipelines/templates/template-job-deploy-azure-env-tag.yml:8]: do we need this here if we're not using it then?

Comment (petrkotas) [reply .pipelines/templates/template-job-deploy-azure-env-tag.yml]: damn, as I said it was late that day already. Good finding. I have removed it. Thanks :)

**Review by bennerv — APPROVED** [review]
LGTM - thanks for making the changes :)

**Pull request merged**